### ISSUE TITLE
could not create/update output branch: rpc error: code = Unknown desc = etcdserver: request is too large

### ISSUE THREAD

**Issue by benwbooth** [pachyderm/pachyderm#3914]
**What happened?**:

Ran `pachctl update pipeline -f -` with the following input:

```
{
  "enable_stats": true,
  "input": {
    "pfs": {
      "glob": "/*/",
      "lazy": true,
      "name": "iarpa",
      "repo": "iarpa-assembly-input-shuffle"
    }
  },
  "parallelism_spec": {
    "coefficient": 1
  },
  "pipeline": {
    "name": "iarpa-wtdbg2-assembly"
  },
  "pod_patch": "[{\"op\": \"replace\", \"path\": \"/containers/0/imagePullPolicy\", \"value\": \"Always\"}]",
  "resource_requests": {
    "cpu": 24
  },
  "standby": true,
  "transform": {
    "cmd": [
      "bash"
    ],
    "image": "bdgp/wtdbg2",
    "stdin": [
      "set -eux -o pipefail\n[[ -d $iarpa ]] || exit\n[[ ! -e \"$iarpa\"/info/family || $(cat \"$iarpa\"/info/family) = bacteria ]] && exit\n# get the median genome size in nucleotides\n# do pass-only and pass/fail assemblies\ng=$( (find \"$iarpa\"/reference_genome/*.{fa,fasta} ||true) |xargs -d \\\\n perl -0076 -ne 'chomp; next if !$_; ($h,$s)=split/\\n/,$_,2; $s=~s/\\s+//g; $s{$ARGV}+=length($s); END{@f=sort {$s{$a}<=>$s{$b}} keys %s; $f=$s{$f[int(@f/2)]}; print \"$f\\n\"}') || true\ng=${g:-125m}\n# set prefix\nmkdir -p \"/pfs/out/$(basename \"$iarpa\")/assembly/wtdbg2\"\nprefix=/pfs/out/$(basename \"$iarpa\")/assembly/wtdbg2/wtdbg2\nwtdbg2 -x ont -g \"$g\" $(printf -- \"-i %q \" \"$iarpa\"/nanopore/*/*/*/*.fastq) -t0 -fo \"$prefix\"\nwtpoa-cns -t0 -i \"$prefix.ctg.lay.gz\" -fo \"$prefix.ctg.fa\"\n# polish with short reads\nbwa index \"$prefix.ctg.fa\"\nfind \"$iarpa\"/illumina/*/*_R2_*.fastq.gz |perl -pe 's/_R2_/_R1_/' |xargs -d \\\\n gzip -cd |pigz >read1.fq.gz\nfind \"$iarpa\"/illumina/*/*_R2_*.fastq.gz |xargs -d \\\\n gzip -cd |pigz >read2.fq.gz\nbwa mem -t 24 \"$prefix.ctg.fa\" read{1,2}.fq.gz |samtools view -Sbu - |samtools sort -@ 24 - |samtools view - |wtpoa-cns -t0 -x sam-sr -d \"$prefix.ctg.fa\" -i - -fo \"$prefix.ctg.polished.fa\"\nmv -v \"$prefix.ctg.polished.fa\" \"/pfs/out/$(basename \"$iarpa\")/assembly/wtdbg2_assembly.fasta\"\n"
    ]
  }
}
```

Received the following error:

```
Reading from stdin.
could not create/update output branch: rpc error: code = Unknown desc = etcdserver: request is too large
```

**What you expected to happen?**:

The pipeline should have been updated.

**How to reproduce it (as minimally and precisely as possible)?**:

Not sure how to reproduce, seems to happen sporadically. Usually I end up having to delete and re-create all pipelines to work around.

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:40:16Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Pachyderm CLI version (use `pachctl version`):
```
COMPONENT           VERSION
pachctl             1.9.0
pachd               1.9.0
```

- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):
```
on-premises
```

- OS (e.g. from /etc/os-release):
```
Arch linux
```

- Others:

**Comment by benwbooth** [comment]
I found another workaround: I edited the etcd kubernetes yaml command section and added the following argument: `--max-request-bytes 33554432`

**Comment by JoeyZwicker** [comment]
@benwbooth Sorry you're hitting this! There are a few different places where we can hit a transaction size limit -- we thought we got most of them, but clearly missed a few. 

In think this is a similar issue to #3879. Shoudn't be too hard to fix and glad you hae a temporary workaround for now.

**Comment by jdoliner** [comment]
Is the branch you're taking as input `iarpa-assembly-input-shuffle@master` the result of a long chain of pipelines? That's the only thing I could think of that would lead to a request that's too large, the content of the pipeline shouldn't have any effect because that code gets stored in a commit, not in etcd.

**Comment by jdoliner** [comment]
This has probably been solved by #3958, which shipped in 1.9.3, feel free to reopen if you hit it again.
